src/functions/index.ts

```diff
@@ -121,10 +152,37 @@ function addInputHandling(
         );
       }
     }
-    return descr.compute.apply(this, args);
+
+    if (countVectorizableCol === 1 && countVectorizableRow === 1) {
+      return compute.apply(this, args);
+    }
+
+    const argsVector: (i: number, j: number) => Arg[] = (i, j) =>
+      args.map((arg, index) => {
+        switch (vectorArgsType[index]) {
+          case "matrix":
+            return arg![i][j];
+          case "horizontal":
+            return arg![i][0];
+          case "vertical":
+            return arg![0][j];
+          case "scalar":
+            return arg;
+        }
+      });
+
+    return generateMatrix(countVectorizableCol, countVectorizableRow, (col, row) => {
+      if (col > vectorizableColLimit - 1 || row > vectorizableRowLimit - 1) {
+        return notAvailableError;
+      }
+      const matrixElm = compute.apply(this, argsVector(col, row));
+      // to avoid mismatching behavior when user try to vectorize arguments on formula that spread
```

Comment:
I'm not sure to understand what this means

Reply:
// In the case where the user tries to vectorize arguments of an array formula, we will get an
// array for every combination of the vectorized arguments, which will lead to a 3D matrix and
// we won't be able to return the values.
// In this case, we keep the first element of each spreading part, just as Excel does, and
// create an array with these parts.
// For exemple, we have MUNIT(x) that return an unitary matrix of x*x. If we use it with a
// range, like MUNIT(A1:A2), we will get two unitary matrixes (one for the value in A1 and one
// for the value in A2). In this case, we will simply take the first value of each matrix and
// return the array [First value of MUNIT(A1), First value of MUNIT(A2)].